cmd/contour/serve.go

```diff
@@ -631,14 +635,15 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		select {
 		case sig := <-c:
 			log.WithField("context", "sigterm-handler").WithField("signal", sig).Info("shutting down")
+			cancel()
```

Comment:
If you make the above changes, then you shouldn't need to explicitly cancel here since the workgroup will handle it for you when this function returns.